lib/pos_blockchain.go

```diff
@@ -375,51 +539,41 @@ func (bc *Blockchain) getLineageFromCommittedTip(desoBlock *MsgDeSoBlock) ([]*Bl
+func (bc *Blockchain) addBlockToBlockIndex(desoBlock *MsgDeSoBlock, blockStatus BlockStatus) error {
 	hash, err := desoBlock.Hash()
 	if err != nil {
 		return errors.Wrapf(err, "addBlockToBlockIndex: Problem hashing block %v", desoBlock)
 	}
 	// Need to get parent block node from block index
 	prevBlock := bc.blockIndex[*desoBlock.Header.PrevBlockHash]
-	// TODO: What should the block status be here? Validated? What combo is correct? Need to check in with Diamondhands.
-	bc.blockIndex[*hash] = NewPoSBlockNode(prevBlock, hash, uint32(desoBlock.Header.Height), desoBlock.Header, StatusHeaderValidated|StatusBlockValidated, UNCOMMITTED)
+	bc.blockIndex[*hash] = NewPoSBlockNode(prevBlock, hash, uint32(desoBlock.Header.Height), desoBlock.Header, blockStatus)
 
-	bc.uncommittedBlocksMap[*hash] = desoBlock
 	return nil
 }
 
-// tryApplyNewTip attempts to apply the new tip to the best chain. It will do the following:
-// 1. Check if we should perform a reorg. If so, it will handle the reorg. If reorging causes an error, return false and error.
-// 2. Check if the incoming block extends the chain tip after reorg. If not, return false and nil
-// 3. If the incoming block extends the chain tip, we can apply it by calling addBlockToBestChain. Return true and nil.
-func (bc *Blockchain) tryApplyNewTip(newTip *MsgDeSoBlock, currentView uint64, lineageFromCommittedTip []*BlockNode) (_appliedNewTip bool, _err error) {
-	// Create a new block node for the new tip
-	newTipBlockNode, err := bc.msgDeSoBlockToNewBlockNode(newTip)
-	if err != nil {
-		return false, errors.Wrapf(err, "tryApplyNewTip: Problem creating new block node: ")
+// tryUpdateBestChain attempts to add the BlockNode passed in to the best chain. It assumes the BlockNode
+// has already been added to the blockIndex and stored in the DB.
+//
+// FIXME: Do we need the bestHeaderChain and the bestHeaderChainMap anymore? I think we need it for initial
+// sync but not sure. Maybe we can delete it? This is a question to revisit when redoing server.go
+func (bc *Blockchain) tryUpdateBestChain(
+	blockNode *BlockNode, currentView uint64, lineageFromCommittedTip []*BlockNode) (_err error) {
+
+	// In the happy path, the block's parent will be the last block in the best chain.
+	uncommittedTip := bc.GetBestChainTip()
+	if uncommittedTip.Hash.IsEqual(blockNode.Parent.Hash) {
+		bc.addBlockToBestChain(blockNode)
+		return nil
 	}
 
-	// Check if the incoming block extends the chain tip. If so, we don't need to reorg
-	// and can just add this block to the best chain.
-	chainTip := bc.GetBestChainTip()
-	if chainTip.Hash.IsEqual(newTip.Header.PrevBlockHash) {
-		bc.addBlockToBestChain(newTipBlockNode)
-		return true, nil
-	}
-	// Check if we should perform a reorg here.
-	// If we shouldn't reorg AND the incoming block doesn't extend the chain tip, we know that
-	// the incoming block will not get applied as the new tip.
-	if !bc.shouldReorg(newTip, currentView) {
-		return false, nil
-	}
```

Comment:
Why are we getting rid of this shouldReorg call? This is critical to fast hotstuff if I'm not mistaken.